bsp/usb.c

```diff
@@ -679,8 +679,14 @@ void Fx3UsbInit(const struct Fx3UsbCallbacks *callbacks)
 		(1UL << FX3_IRQ_GCTL_POWER) | (1UL << FX3_IRQ_USB_CORE) | (1UL << FX3_IRQ_GCTL_CORE));
 }
 
-void Fx3UsbEnableInEndpoint(uint8_t ep, Fx3UsbEndpointType_t type, uint16_t pktsize)
+void Fx3UsbEnableInEndpoint(uint8_t ep, Fx3UsbEndpointType_t type, Fx3UsbSpeed_t s)
```

Comment:
This is a bit weird.  Why shouldn't you be able to set a smaller pktsize than the max?  I think it's better to keep the BSP API as it is (and possibly add a chek that the pktsize is not too large) and do the selection of 512 or 1024 byte packets in  main.c instead.

Reply:
> Why shouldn't you be able to set a smaller pktsize than the max? 
USB 2 bulk endpoint support 512 , and USB 3 one 1024.
i will put check for speed in main.c file's SetupData function.

Reply:
What I mean is that they both support _smaller_ than 512, so if you want to have a packet size of, say, 256, you should be able to get it.  Which is not possible if you remove the pktsize argument.